src/orderer.ts

```diff
@@ -0,0 +1,109 @@
+
+import { writeBlocks } from './writer';
+import * as fs from 'fs';
+import * as tfTypes from './terraformTypesEnum';
+
+export type ElementIndex = {
+    element: any,
+    position: number
+};
+
+export class Orderer {
+
+    protected typesMap: Map<string, Array<number>>;
+
+    constructor(typesMap: Map<string, Array<number>>) {
+        this.typesMap = typesMap;
+    }
+
+    public segregateOrderedTypes(arrOrigin: any[]): Map<string, Array<ElementIndex>> {
+        let orderList = new Map<string, Array<ElementIndex>>();
+        Orderer.elementsToFetch().forEach(el => {
+            orderList.set(el.element,
+                this.orderArrayByKey(el.element, arrOrigin, el.position));
+        });
+
+        return orderList;
+    }
+
+    public static elementsToFetch(): ElementIndex[] {
+        return [
+            {
+                element: tfTypes.elementTypeEnum.DATA,
+                position: 2
+            },
+            {
+                element: tfTypes.elementTypeEnum.VARIABLE,
+                position: 1
+            },
+            {
+                element: tfTypes.elementTypeEnum.RESOURCE,
+                position: 2
+            }
+        ];
+    }
+
+    protected orderArrayByKey(elementKey: string, arrOrigin: any[], elementSortIndex: number): any[] {
+        let resultList: any[] = [];
+        if (this.typesMap.has(elementKey)) {
+            resultList = this.sortListByTFElement(
+                this.iterateElementArray(elementKey, arrOrigin),
+                elementSortIndex
+            );
+        }
+
+        return resultList;
+    }
+
+    protected sortListByTFElement(unsortedList: any[], index: number): any[] {
```

Comment:
```suggestion
    protected sortListByTfElement(unsortedList: any[], index: number): any[] {
```
This is just a question I have, which convention should we adopt?
Also a link for the discussion
https://stackoverflow.com/questions/15526107/acronyms-in-camelcase

Reply:
Good point @m-yassuo I didn't notice it before! I will fix it!